### PR TITLE
chore: release `v1.0.0-alpha.2`

### PR DESCRIPTION
**Pull request by sergio-mena**
Changes in preparation for our second alpha release of `v1.0.0`:

- [CHANGELOG rendered](https://github.com/cometbft/cometbft/blob/release/v1.0.0-alpha.2/CHANGELOG.md)
- [Manual E2E run](https://github.com/cometbft/cometbft/actions/runs/8283569894)

IMPORTANT:

The first commit of this PR is manually undoing the `unclog release` that was performed for `v1.0.0-alpha.1` (#1710). In our release process, pre-releases do not have their own section in the changelog, so we must _not_ run `unclog release` when cutting pre-releases (see `RELEASES.md`).

Therefore, it is advised to exclude the first commit from the diff view when reviewing this PR.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: UPGRADING.md [hunk 1]
```diff
@@ -29,6 +28,36 @@ versioning:
...
+called Proposer-Based Timestamps (PBTS).
+The existing algorithm, called BFT-Time is kept for backwards compatibility.
+Upgrading to `v1.0` does not automatically switch the chain from BFT-Time
+to PBTS; rather a ConsensusParam called `PbtsEnableHeight` can be set to a future
+height to transition from BFT-Time to PBTS.
+This flexible mechanism allows chains disentagle the upgrade to `v1.0` from the transition
+in the algorithm used for block times.
+For further information, please check the [PBTS specification][pbts-spec].
+
+#### ABCI Mutex
```
Comment (sergio-mena) [inline UPGRADING.md:43]:
```suggestion
### ABCI Mutex
```

**Comment by hvanz** [comment]
#2327 should be part of the QA process, so we need to backport it to v1.x and include it in this release.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # CHANGELOG
 
-## v1.0.0-alpha.1
+## Unreleased
 
 *December 4, 2023*
```
Comment (hvanz) [inline CHANGELOG.md:5]: Do we need to update this date?

Comment (sergio-mena) [reply CHANGELOG.md:5]: We can deal with this when cutting the final `v1.0.0`

**Review by hvanz — APPROVED** [review]

**Pull request merged**